src/Mod/PartDesign/Gui/ViewProviderDatum.cpp

```diff
@@ -101,6 +101,11 @@ ViewProviderDatum::~ViewProviderDatum()
 
 void ViewProviderDatum::attach(App::DocumentObject *obj)
 {
+    if (auto geo = dynamic_cast<App::GeoFeature*>(obj)) {
+        App::Material mat = geo->getMaterialAppearance();
+        mat = ShapeAppearance[0];
+        geo->setMaterialAppearance(mat);
+    }

```

Comment:
Not sure about this section. You're assigning mat different values on lines 105 and 106. Only the second assignment would have any effect. This occurs in multiple places

Reply:
This is needed because datum features, shape binders and sub-shape binders have different default colours. Without this change they are light grey and opaque instead of yellowish and 60% transparency.

This colour is set inside the constructors of the view providers and would be overridden by the material of the features

Reply:
I think it would be smart to add a comment to that effect since it's non-obvious. Someone refactoring in the future might be tempted to remove the first assignment

Reply:
Updated PR.